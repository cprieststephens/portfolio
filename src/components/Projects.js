import React from "react";
import { Link } from "react-router-dom";
import cocktailCompanion from "../img/cocktail-companion.jpg";
import surrealEstate from "../img/surreal-estate.jpg";
import weatherApp from "../img/weather-app.jpg";
import "../styles/projects.css";

const Projects = () => {
  return (
    <div className="projects">
      <div className="projects-item">
        <h1>Cocktail Companion</h1>
        <Link to="https://www.cocktailcompanion.bar/" target="_blank">
          <img
            className="projects-item__image"
            src={cocktailCompanion}
            alt="Cocktail Companion screenshot"
          />
        </Link>
        <p>
          Collaboratively built with React, JavaScript, CSS, Jest and React
          Testing Library.
        </p>
        <p>
          A cocktail mixing app that displays API data and allows users to
          search for drinks by name, filter by spirit and view ingredients and
          instructions.
        </p>
        <p>
          <Link
            className="projects-item__repo"
            to="https://github.com/josenymad/cocktail-companion"
            target="_blank"
          >
            View the repo here
          </Link>
          <Link
            className="projects-item__repo"
            to="https://www.youtube.com/watch?v=9MXMxw37I-k"
            target="_blank"
          >
            <span> and presentation here.</span>
          </Link>
        </p>
      </div>
      <div className="projects-item">
        <h1>Surreal Estate</h1>
        <img
          className="projects-item__image"
          src={surrealEstate}
          alt="Surreal Estate screenshot"
        />
        <p>
          Built with React, JavaScript, CSS, Firebase, Jest and React Testing
          Library.
        </p>
        <p>
          A property listing app that allows users to view property data via an
          API. Signed in users can also add and remove data. This project is
          currently in progress.
        </p>
        <p>
          <Link
            className="projects-item__repo"
            to="https://github.com/cprieststephens/surreal-estate"
            target="_blank"
          >
            View the repo here.
          </Link>
        </p>
      </div>
      <div className="projects-item">
        <h1>Weather App</h1>
        <Link to="https://check-the-forecast.netlify.app/" target="_blank">
          <img
            className="projects-item__image"
            src={weatherApp}
            alt="Weather App screenshot"
          />
        </Link>
        <p>
          Built with React, JavaScript, CSS, Jest and React Testing Library.
        </p>
        <p>
          A weather app that allows users to search by city to fetch forecast
          data from an API. Users can also access a more detailed forecast for
          each date by clicking a button.
        </p>
        <p>
          <Link
            className="projects-item__repo"
            to="https://github.com/cprieststephens/weather-app"
            target="_blank"
          >
            View the repo here.
          </Link>
        </p>
      </div>
    </div>
  );
};

export default Projects;
