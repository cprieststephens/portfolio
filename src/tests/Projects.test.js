import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Projects from "../components/Projects";

describe("Projects", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Projects />
      </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct titles", () => {
    render(
      <MemoryRouter>
        <Projects />
      </MemoryRouter>,
    );
    const cocktailCompanion = screen.getByText(/cocktail companion/i);
    const surrealEstate = screen.getByText(/surreal estate/i);
    const weatherApp = screen.getByText(/weather app/);

    expect(cocktailCompanion).toBeInTheDocument();
    expect(surrealEstate).toBeInTheDocument();
    expect(weatherApp).toBeInTheDocument();
  });

  it("renders the correct images", () => {
    render(
      <MemoryRouter>
        <Projects />
      </MemoryRouter>,
    );
    const cocktailCompanionImage = screen.getByAltText(
      "Cocktail Companion screenshot",
    );
    const surrealEstateImage = screen.getByAltText("Surreal Estate screenshot");
    const weatherAppImage = screen.getByAltText("Weather App screenshot");

    expect(cocktailCompanionImage).toBeInTheDocument();
    expect(surrealEstateImage).toBeInTheDocument();
    expect(weatherAppImage).toBeInTheDocument();
  });
});
