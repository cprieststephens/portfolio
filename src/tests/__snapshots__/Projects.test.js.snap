// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects renders correctly 1`] = `
<DocumentFragment>
  <div
    class="projects"
  >
    <div
      class="projects-item"
    >
      <h1>
        Cocktail Companion
      </h1>
      <a
        href="https://www.cocktailcompanion.bar/"
        target="_blank"
      >
        <img
          alt="Cocktail Companion screenshot"
          class="projects-item__image"
          src="cocktail-companion.jpg"
        />
      </a>
      <p
        class="projects-item__description"
      >
        A cocktail mixing app that displays API data and allows users to search for drinks by name, filter by spirit and view ingredients and instructions.
      </p>
      <p
        class="projects-item__description"
      >
        Collaboratively built with React, JavaScript, CSS, Jest and React Testing Library.
      </p>
      <p
        class="projects-item__description"
      >
        <a
          class="projects-item__repo"
          href="https://github.com/josenymad/cocktail-companion"
          target="_blank"
        >
          View the repo here
        </a>
        <a
          class="projects-item__repo"
          href="https://www.youtube.com/watch?v=9MXMxw37I-k"
          target="_blank"
        >
          <span>
             and presentation here.
          </span>
        </a>
      </p>
    </div>
    <div
      class="projects-item"
    >
      <h1>
        Surreal Estate
      </h1>
      <img
        alt="Surreal Estate screenshot"
        class="projects-item__image"
        src="surreal-estate.jpg"
      />
      <p
        class="projects-item__description"
      >
        A property listing app that allows users to view property data via an API. Signed in users can also add and remove data. This project is currently in progress.
      </p>
      <p
        class="projects-item__description"
      >
        Built with React, JavaScript, CSS, Firebase, Jest and React Testing Library.
      </p>
      <p
        class="projects-item__description"
      >
        <a
          class="projects-item__repo"
          href="https://github.com/cprieststephens/surreal-estate"
          target="_blank"
        >
          View the repo here.
        </a>
      </p>
    </div>
    <div
      class="projects-item"
    >
      <h1>
        Weather App
      </h1>
      <a
        href="https://check-the-forecast.netlify.app/"
        target="_blank"
      >
        <img
          alt="Weather App screenshot"
          class="projects-item__image"
          src="weather-app.jpg"
        />
      </a>
      <p
        class="projects-item__description"
      >
        A weather app that allows users to search by city to fetch forecast data from an API. Users can also access a more detailed forecast for each date by clicking a button.
      </p>
      <p
        class="projects-item__description"
      >
        Built with React, JavaScript, CSS, Jest and React Testing Library.
      </p>
      <p
        class="projects-item__description"
      >
        <a
          class="projects-item__repo"
          href="https://github.com/cprieststephens/weather-app"
          target="_blank"
        >
          View the repo here.
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
